### PR TITLE
feat: add slack notification for users in debt

### PR DESCRIPTION
**Pull request by hkopij**

### CODE REVIEW

File: app/services/slack_id_importer.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class SlackIdImporter
+  def initialize(user)
+    @user = user
+  end
+
+  def perform
+    if @user.slack_id == nil
```
Comment (ghost) [inline app/services/slack_id_importer.rb]: WDYTA `unless @user.slack_id`?

File: app/workers/slack_notifier_worker.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+class SlackNotifierWorker
+  include Sidekiq::Worker
+
+  def perform
+    User.all.active.each do |user|
+      if user.slack_id && user.total_debt.to_i < -30
```
Comment (ghost) [inline app/workers/slack_notifier_worker.rb]: WDYTA moving `-30` into a const?

File: spec/workers/slack_id_import_worker_spec.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+RSpec.describe SlackIdImportWorker, type: :worker do
+  let(:time) { (Time.zone.today + 1.hours).to_datetime }
+  let(:scheduled_job) { described_class.perform_at(time, 'Awesome', true) }
+
+  context 'occurs weekly' do
+    it 'occurs at expected time' do
+      scheduled_job
+      assert_equal true, described_class.jobs.last['jid'].include?(scheduled_job)
+      expect(described_class).to have_enqueued_sidekiq_job('Awesome', true)
```
Comment (ghost) [inline spec/workers/slack_id_import_worker_spec.rb]: Your spec says `occurs at expected time`, so I'd probably check if it _really_ does occur at the expected time, not just that it _occurs_. WDYTA adding `.at(time)` on LOC 14 (https://github.com/philostler/rspec-sidekiq#have_enqueued_sidekiq_job)?

File: app/services/slack_id_importer.rb [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+class SlackIdImporter
+  def initialize(user)
+    @user = user
+  end
+
+  def perform
+    if @user.slack_id.blank?
+      url = "https://slack.com/api/users.lookupByEmail?email=#{@user.email}&token=#{ENV['SLACK_OAUTH_TOKEN']}"
```
Comment (ghost) [inline app/services/slack_id_importer.rb]: Nit: WDYTA formatting it so that it doesn't exceed the 80 char limit?

**Review by ghost — APPROVED** [review]

File: spec/workers/slack_id_import_worker_spec.rb (hunk 3 above)
Comment (hkopij) [reply spec/workers/slack_id_import_worker_spec.rb]: True, spec says "occurs at expected time", but test is checking something else - if the worker is scheduled. Will fix that

**Pull request merged**